TheJumpCloud/support
# SA-3184 - Radius Cert Repo Cleanup

## Issues
* [SA-3184](https://jumpcloud.atlassian.net/browse/SA-3184) - Radius Cert Repo Cleanup

## What does this solve?
The purpose of this is to reduce the clutter within the file structure and to break out functions into their individual files to make future work easier to find/understand

## Is there anything particularly tricky?
A lot of the scripts and functions make use of the `$PSScriptRoot` variable, which references the path of that individual script run. This was fine when everything was under the same folder, however; it does not work when the functions are relegated to their own individual files.

To fix this, a global variable is created when `Start-JCDeployment.ps1` is executed that saves the path of `../support/scripts/automation/radius`

There was also an issue when generating the UserCerts that relied on the variables set in Config.ps1. This was remedied by calling `. "$JCScriptRoot/config.ps1"` in the begin block of `Generate-UserCert.ps1`

## How should this be tested?

1. Pull Branch
2. Edit `Config.ps1` and update variables for your org
3. Run `./Start-RadiusDeployment.ps1`
4. Ensure the following:
    * Root Certs are generated correctly (`../Radius/Cert`)
    * User Certs are generated correctly in their proper folder (`../Radius/UserCerts`) 
    * users.json file is generated correctly
    * Commands are generated correctly and the correct files are uploaded to them
    * Reporting is accurate for your environment


[SA-3184]: https://jumpcloud.atlassian.net/browse/SA-3184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
